report/sections/TestRunner/chapter.tex

```diff
@@ -1,33 +1,32 @@
 \chapter{Test Runner} \label{chap:TestRunner}
-In chapter \ref{chap:TestRunner}, we described how running multiple GHC interpreter processes asynchronously eventually crashes the Test Runner web server given sufficiently many simultaneous requests.
-As mentioned, we needed a way to limit the number of running GHC interpreter processes in order to prevent crashes.
-
-The Test Runner exposes an endpoint \texttt{/haskell/submit/}, which is used to post code submissions.
-Originally, we used the Rust crate Rocket to implement this endpoint.
-However, we eventually switched to using the axum crate instead, as it is developed and maintained by the same developers behind the Tokio package we make heavy use of for asynchronous programming and multithreading.
-Furthermore, it was deemed easier to implement a shared state among endpoints by leveraging axum's \texttt{layer} feature.
-We give an overview and a more technical explanation of how this was done in section \ref{sec:queue-system}.
-
-When the Test Runner receives the request from Next.js, its singular goal is to run the tests against the code and send the result of this back as fast as possible.
-In order to run the tests, we need to run the GHC interpreter on the test code, which, in turns, need access to the exercise submission code.
+When the Test Runner receives the request from the backend component, its singular goal is to run the tests against the code and send back the result as fast as possible.
```

Comment:
```suggestion
When the Test Runner receives the request from the backend component, its singular goal is to run the tests against the code and make the result available as fast as possible.
```
We don't send the results back without pollling, and that violates the goal criteria otherwise.

Reply:
```suggestion
When the Test Runner receives the request from the backend component, its singular goal is to run the tests on the submitted code and make the result available as fast as possible.
```